### DISCUSSION TITLE
Missing logfile - can't get kimai (docker) to write any logs

### DISCUSSION THREAD

**Discussion by dreiekk** [tobybatch/kimai2#431] [Q&A] [answered]
Hi there, maybe someone can point me in the right direction...

I'm currently developing a kimai plugin and have trouble getting uncaught php errors on our production system logged.
The server is running Kimai `v1.28.1` using the `kimai/kimai2:apache-1.28.1-prod` docker image.

The `System > Doctor > Logfile (max. 100 last lines)` page displays the message `Missing logfile`.

Both directories `/opt/kimai/var/log` and `/opt/kimai/var/logs` in side the docker container are empty.
They seem to have the right permissions set:
```
www-data@24a90d79997c:/opt/kimai/var$ ls -ld log*
drwxr-xr-x 2 www-data www-data 4096 Nov 10 19:35 log
drwxrwxrwx 2 www-data www-data 4096 Oct 30 11:40 logs
```

When I raise a php error in my plugin, I get the following error in my browser as expected:
![grafik](https://user-images.githubusercontent.com/25348281/201710490-b805c902-af13-4085-b367-f511f0484dbb.png)

I tried to change the `level` to `debug` in `/opt/kimai/config/packages/prod/monolog.yaml`:
```yaml
monolog:
  handlers:
    main:
      type: error_log
      level: debug
```
and reloaded kimai with `bin/console kimai:reload`, restarted the container, but still no logs are written.

I raised an "Internal server error" similiar to the screenshot above by stopping a timesheet twice by opening two tabs as described here:
https://github.com/kevinpapst/kimai2/issues/1730#issuecomment-633134426_
But there are still no logs written.

Maybe I got something about logging in kimai fundamentally wrong - any help appreciated.
Feel free to ask for any information regarding the kimai installation.

**Comment by kevinpapst** [comment, marked as answer]
Docker images are not writing to the logfile, but to stdout => the docker log output (don't know how that is called).

Developing in prod environment is a terrible idea. Setup the dev image with APP_ENV

**Reply by dreiekk** [reply]
> Docker images are not writing to the logfile, but to stdout => the docker log output

Thanks! That's the thing I was missing. No wonder I didn't find anything file-based.
I thought something was broken due to the `Missing logfile` message on the Doctor page.

> Developing in prod environment is a terrible idea.

But it's much more fun to test in production!~ kappa
There was a bug I couldn't reproduce in my local dev environment, so I wanted to inspect the logs of our prod system.
I setup anonther prod container locally to see if I can get any logs out of it.

**Reply by kevinpapst** [reply]
> But it's much more fun to test in production!~ kappa

Hahaha, the adrenalin rush is much better^^

**Comment by tobybatch** [comment]
Better still use a dev tag:

```
docker pull kimai/kimai2:apache-dev
```

or

```
docker pull kimai/kimai2:fpm-dev
```

And then use `docker attach CONTAINER_NAME`

I can't help you with better instructions because I have no idea how you are start the container, or what OS.

**Reply by dreiekk** [reply]
Thanks for the quick reply!
I was mistakenly expecting a logfile instead of `docker logs`

**Comment by jassonmc** [comment]
Sorry for resurrecting an old post, but is it still true that Kimai installed with Docker does not produce any logs in /var/log/prod.log ?
If so, is there eventually any manual work around for it?
Would be awesome if potential errors could already be seen in the Kimai GUI in the Doctor.

**Comment by tobybatch** [comment]
The containers should be logging to standard out. If they logged to file they would eventually fill up your disk and you'd never know why.

Try attaching to the container:

```
docker compose logs -f CONTAINER_NAME
```

or

```
docker logs -f CONTAINER_NAME
```
